ihmc-pub-sub/src/main/java/us/ihmc/pubsub/attributes/ParticipantProfile.java

```diff
@@ -136,112 +136,136 @@ public ParticipantProfile discoveryServer(String discoveryServerAddress, int dis
-         TransportDescriptorType transportDescriptor = new TransportDescriptorType();
-         transportDescriptor.setTransportId(transportName);
-         transportDescriptor.setType("UDPv4");
 
-         TransportDescriptorType.InterfaceWhiteList addressWhitelist = new InterfaceWhiteList();
+      // Create userTransports if it doesn't exist
+      if (profileType.getRtps().getUserTransports() == null)
+         profileType.getRtps().setUserTransports(new UserTransports());
 
-         for (InetAddress addr : bindToAddressRestrictions)
+      // Add to userTransports if it doesn't exist
+      {
+         boolean existsInUserTransports = false;
+         for (String transportId : profileType.getRtps().getUserTransports().getTransportId())
          {
-            JAXBElement<String> addressElement = new JAXBElement<>(new QName(FastRTPSDomain.FAST_DDS_XML_NAMESPACE, "address"), String.class, addr.getHostAddress());
-            addressWhitelist.getAddressOrInterface().add(addressElement);
+            if (transportId.equals(transport.getTransportId()))
+            {
+               existsInUserTransports = true;
+               break;
+            }
          }
-         
-         transportDescriptor.setInterfaceWhiteList(addressWhitelist);
-         addTransport(transportDescriptor);
+         if (!existsInUserTransports)
+            profileType.getRtps().getUserTransports().getTransportId().add(transport.getTransportId());
       }
-      
-      return this;
-   }
 
-   /**
-    * Add a transport to use with this participant and register it to this participant
-    * 
-    * @param transport
-    * @return
-    */
-   public ParticipantProfile addTransport(TransportDescriptorType transport)
-   {
-      transportDescriptors.getTransportDescriptor().add(transport);
-      
-      if(profileType.getRtps().getUserTransports() == null)
-      {
-         profileType.getRtps().setUserTransports(new ParticipantProfileType.Rtps.UserTransports());
-      }
-      profileType.getRtps().getUserTransports().getTransportId().add(transport.getTransportId());
-      
       return this;
    }
    
    /**
     * Add a shared memory transport to this participant.
-    * 
     * By setting useBuiltinTransports to false, you can use only a shared memory transport
-    *  
-    * @return
     */
    public ParticipantProfile addSharedMemoryTransport()
    {
-      String transportName = UUID.randomUUID().toString();
       TransportDescriptorType transportDescriptor = new TransportDescriptorType();
-      transportDescriptor.setTransportId(transportName);
+      transportDescriptor.setTransportId(UUID.randomUUID().toString());
       transportDescriptor.setType("SHM");
-      
+
       addTransport(transportDescriptor);
-      
+
       return this;
    }
-   
-   public ParticipantProfile useBuiltinTransports(boolean useBuiltinTransports)
+
+   public ParticipantProfile addUDPv4Transport(InetAddress... addressRestriction)
```

Comment:
I added this method, the logic was duplicated in a couple places. It adds a UDPv4 transport with optional restrictions on addresses